src/main/java/school/hei/haapi/authentication/AwsCognitoJwtAuthFilter.java

```diff
@@ -0,0 +1,32 @@
+package school.hei.haapi.authentication;
+
+import javax.servlet.FilterChain;
+import javax.servlet.GenericFilter;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import lombok.SneakyThrows;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AwsCognitoJwtAuthFilter extends GenericFilter {
+
+  private final AwsCognitoJwtIdToken cognitoIdTokenProcessor;
+
+  public AwsCognitoJwtAuthFilter(final AwsCognitoJwtIdToken cognitoIdTokenProcessor) {
+    this.cognitoIdTokenProcessor = cognitoIdTokenProcessor;
+  }
+
+  @SneakyThrows
```

Comment:
I used to use SneakyThrows. But now I think it's not a good idea to use it in prod. Please remove SneakyTrhows

Reply:
The method doFilter is an override method that doesn't throw the authentication Exceptions (NotAuthorizedException, ParseException, BadJOSEException, JOSEException)
So either I use sneakyThrows or i try/catch the exception.